src/main/java/com/evgeniyfedorchenko/hogwarts/controllers/StudentController.java

```diff
@@ -18,15 +19,16 @@ public StudentController(StudentService studentService) {
     }
 
     @PostMapping
-    public ResponseEntity<Student> createStudent(@RequestBody Student student) {
-        return ResponseEntity.ok(studentService.createStudent(student));
+    public Student createStudent(@RequestBody Student student) {
+        return studentService.createStudent(student);
     }
 
     @GetMapping(path = "/{id}")
     public ResponseEntity<Student> getStudent(@PathVariable Long id) {
-        return studentService.getStudent(id)
+        return ResponseEntity.ok(studentService.getStudent(id).get());
```

Comment:
```suggestion
        return ResponseEntity.of(studentService.getStudent(id));
```
не нужно никакой get, просто вернуть Optional в of и всё